### DISCUSSION TITLE
Can't connect to icecast2

### DISCUSSION THREAD

**Discussion by ourtechaus** [fayaaz/mixxx-pi-gen#42] [Q&A] [answered]
I'm having issues with Live Broadcasting - **No Error - Please check your connection to the internet and verify that your username and password are correct**

I can connect to the same icecast2 server from a Windows or Ubuntu PC running Mixxx, but not from this Mixxx/RPI distro. 

Does anyone have any tricks to get it to talk, perhaps the settings I am using needs to be slightly modified on the MixxxRPI

Type: icecast2
Host: using local IP
Mount: /stream
Login: source
Password: nothing works!
Format: Ogg vorbis or MP3

It's so straight forward on the working devices I can't help but think it's a bug. But what would I know :P

Thanks getting Mixxx on RPI hopefully we can get over this hurdle.

**Comment by fayaaz** [comment]
Not sure if icecast2 support is optionally built in Mixxx - I will have a look at that. 
Do you have any logs from Mixxx and what happens when you try and connect? Are you able to ping the shoutcast ip you are trying to connect to from your Pi?

**Reply by fayaaz** [reply]
I don't know if this is relevant to you: https://mixxx.discourse.group/t/unable-to-connect-to-icecast2-server-socket-error/21780/5

What versions of Mixxx are you running on Windows/Ubuntu?

**Reply by ourtechaus** [reply]
Hi thanks,
At some stage of my troubleshooting I was seeing a socket error, but this didn't persist. Now I consistently get No error.
I believe icecast2 is supported as it works on my other test devices and appears in the dropdown as an option.
Yes, I can ping the shoutcast IP from Mixxx-Pi.

My other test devices that work fine are:
Mixxx version 2.2.4 Windows and 2.3.2 Ubuntu

I extracted this from the logfile
Debug [ShoutOutput 'Icecast2']: ShoutConnection - run: Starting thread
Debug [ShoutOutput 'Icecast2']: ShoutConnection - processConnect
Debug [ShoutOutput 'Icecast2']: ShoutConnection - "Icecast2" : updating from preferences
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Using server URL: QUrl("//source@192.168.1.199:8000/stream")
Debug [ShoutOutput 'Icecast2']: RETURNING SERIAL  385595920
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Debug [Main]: QGLFramebufferObject: Framebuffer incomplete attachment.
Debug [Main]: QGLFramebufferObject: Framebuffer incomplete attachment.
Warning [Main]: GLSLWaveformRendererSignal::createFrameBuffer - frame buffer not valid
Debug [Main]: QGLFramebufferObject: Framebuffer incomplete attachment.
Debug [Main]: QGLFramebufferObject: Framebuffer incomplete attachment.
Warning [Main]: GLSLWaveformRendererSignal::createFrameBuffer - frame buffer not valid
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Debug [ShoutOutput 'Icecast2']: ShoutConnection - Connection pending. Waiting...
Warning [ShoutOutput 'Icecast2']: Streaming server made error: -3
Warning [ShoutOutput 'Icecast2']: ShoutConnection - processConnect() error: -3 "No error"
Debug [ShoutOutput 'Icecast2']: ShoutConnection - processConnect() returning false
Warning [ShoutOutput 'Icecast2']: Streaming error:  "No error\n\nPlease check your connection to the Internet and verify that your username and password are correct."

**Comment by ourtechaus** [comment]
Hello Fayaaz, I am still unable to connect and ready to abandon this which is a shame.  I wish it was as simple as me typing the incorrect password, however after 100s of attempts by me and colleagues with the correct password (definitely correct) - it must have a glitch.

I understand if this is not something you'd able to pursue.

All the best

M

**Reply by fayaaz** [reply, marked as answer]
Hi there, I believe it may be connected to this: https://bugs.launchpad.net/mixxx/+bug/1922049
There is a new version of Debian available which has a different version of libshout - could you try the image here which is running Debian 11:
https://github.com/fayaaz/mixxx-pi-gen/suites/5198089185/artifacts/174088214

Please do let me know how you get on.

**Reply by ourtechaus** [reply]
Hello friend thank you so much, I can now connect to the Icecast2 server perfectly! I am very grateful.. All the best!

**Comment by fayaaz** [comment]
Fixed in newer versions of the image
